### ISSUE TITLE
Front-end bootstrap

### ISSUE THREAD

**Issue by Lapin0t** [tozti/tozti#3]
- [ ] initialize the vue.js application
- [ ] document a way for extensions to patch existing views or register new views

**Comment by Lapin0t** [comment]
Hey, so what's the state of the registration of views? This should be done so that extensions can start writing front-end code without worrying. My guess here would be that there are 3 different component types to register:
- view (full page) components, these components will not be reused by other people and should be registered together with a route with vue-router. Question: how does vue-router work, do you want to simply let people call methods on the global router object? Or do you want to code a few wrappers to make things simpler and more consistent? I probably like the second approach better since in any case we will want to wrap them with a common header, some toolbars etc before passing them to the router. 
- internal components (components defined inside an extension that are not meant to be reused). These components should probably _not_ be registered globally (with `Vue.component`), but only registered locally in the relevent parent (probably a page component).
- public reusable components. Mostly used by the core (and registered globally), but some extensions might want to export reusable components. Maybe we can have the convention that extensions register them globally but namespace the tag name like `"{extension}-{some-name}"`. This could be enforced by providing a wrapper too.
- resource-type specific components (eg the _thumbnail view_ of some resource, or the _full page view_ of some resource). I would go with a 2-part api: extensions can export a new _type-specific class of component_, and they can also register instances of some component-class for a particular resource type. Basic implementation of that kind of idea: https://jsfiddle.net/t9f5bzq6/4/. We would have to provide 2 functions:
  - `register_class(class_name, default_component)` creates a new global component with tag `{class_name}`, it has a single `prop` named `resource` that should contain the id of the resource to display. When the type is unknown (no registered instance for that class+type couple) the `default_component` is used.
  - `register_instance(class_name, type_ident, component)` registers a component for one class+type couple.

Note that it might be a good idea to make everything somewhat declarative, that is, the `index.js` should only export stuff and when we render the `index.html` in the startup we generate a small `bootstrap.js` script that loads everything required and registers stuff.

Would that be ok?

**Comment by Lapin0t** [comment]
I see that the doc now mentions `tozti.store.commit('registerWidget', {...})`. Is this stabilized? I must say i don't like this RPC-like design so much, why not `tozti.store.registerWidget({...})`? Or an unique `tozti.store.registerExtension({widgets: [...], ...})`?

**Comment by liautaud** [comment]
Well, nothing is really stabilized in this area anyway, but I don't think @flupe meant `registerWidget` as "the action to register a widget", but rather as the widget displaying a button to register. The name should probably have been `registrationWidget`, but we don't really care about any of this anyway.

**Comment by flupe** [comment]
No, actually this was removed. The store (as in `vuex` store) will only be used to store entities (or what we are now calling resources?). And we will provide abstraction over this message-passing design for extensions to use. (Which will hide data synchronisation and such). Widgets have nothing to do in this store, so I'm moving them somewhere else. And we probably will as you said just do `tozti.registerWidget(...)`.
